ext/standard/tests/strings/str_truncate.phpt

```diff
@@ -0,0 +1,43 @@
+--TEST--
+Testing str_truncate() function
+--FILE--
+<?php
+
+/* Prototype: string substr( string str, int start[, int length] )
+ * Description: Truncates the string to the new length.
+ */
+
+$string = 'Hello world!';
+
+// Should not take effect, new_len is too big.
+str_truncate($string, strlen($string) + 1);
+var_dump($string);
+// Should not take effect, new_len is less than zero.
+str_truncate($string, -1);
+var_dump($string);
+
+// Should truncate $string to the string "Hello".
+str_truncate($string, strlen('Hello'));
+var_dump($string);
+// Should not take effect, it cannot untruncate back.
+str_truncate($string, strlen('Hello world!'));
+var_dump($string);
+// Should truncate to an empty string.
+str_truncate($string, 0);
+var_dump($string);
+// Should not take effect, it cannot untruncate back.
+str_truncate($string, strlen('Hello'));
+var_dump($string);
+
```

Comment:
Add a test case to ensure no CoW variables are modified unexpectedly.

$string = "Hello World";
$cow = $string;
str_tuncate($string, strlen("Hello"));
var_dump($string); // string(5) "Hello"
var_dump($cow); // string(11) "Hello World"

Spoiler alert; In your patch's current state, this will fail as $cow will have been modified as well.